src/main/java/frc/team2412/robot/subsystems/LauncherSubsystem.java

```diff
@@ -278,7 +293,22 @@ public double getAngleSpeed() {
 		return launcherAngleEncoder.getVelocity();
 	}
 
-	public void setAngleManual(double joystickInput) {
+	public void restoreLimits() {
+		this.ignoreLimits = false;
+	}
+
+	public void setAngleManual(double joystickInput, boolean powerControl, boolean ignoreLimits) {
+		this.ignoreLimits = ignoreLimits;
+		if (powerControl || ignoreLimits) {
+			launcherAngleOneMotor.set(ignoreLimits ? joystickInput * 0.3 : joystickInput);
+			manualAngleSetpoint =
+					MathUtil.clamp(
+							Units.degreesToRotations(getAngle()),
+							PIVOT_SOFTSTOP_BACKWARD,
+							PIVOT_SOFTSTOP_FORWARD);
+			return;
+		}
```

Comment:
Why we're setting `manualAngleSetpoint` when we're using power control?

Reply:
because we want the launcher to not snap back to the previous set point after disabling power control